# CSE110 Assignment 2

This assignment is to guide you to solve basic programming problems using loop in Python. 

## Write the Python code of the following problems:

### Task 1

Write python program, which prints the following sequences of values in loops:

a)	24, 18, 12, 6, 0, -6\
b)	-10, -5, 0, 5, 10, 15, 20\
c)	18, 27, 36, 45, 54, 63\
d)	18,-27, 36,-45,54,-63

**Hints(1):** Use a while loop for solving these problems.

**Hints(2):** We are already familiar with the print() function. But when we use it to print any value it automatically adds an additional newline after each print statement.

For example:\
print(1)\
print(2)

Output:\
1\
2

=====================================================================

To solve this problem, in Python3, we need to add an extra argument (end = " ") in the print function which tells the program to skip printing the additional newline.

For example:\
print(1, end =" ")\
print(2)

Output:(prints the next output right to the previous one)\
12

=====================================================================

In Task-1(a), the loop counter should be initialized at 24 and the loop should terminate when the loop counter reaches -6. The difference between the first two values is 24-18=6. So the loop counter value is getting decremented by 6 in every iteration.

For your understanding task 1(a) code is done for you.


In [1]:
# a) 24, 18, 12, 6, 0, -6

# initialize loop counter
counter = 24

# loop structure
while counter >= -6:
    
    #inside loop body
    if counter == -6:
        print(counter, end = "") 
    else:
        print(counter, end = ", ") 

    counter = counter - 6 #updating loop counter
    #inside loop body
    
#outside loop body

24, 18, 12, 6, 0, -6

In [2]:
# b)-10, -5, 0, 5, 10, 15, 20

for counter in range(-10,20+1,5):
    if counter == 20:
        print(counter, end=" ")
    else:
        print(counter, end=", ")
        

-10, -5, 0, 5, 10, 15, 20 

In [3]:
# c)18, 27, 36, 45, 54, 63

for counter in range(18,63+1,9):
    if counter == 63:
        print(counter, end=" ")
    else:
        print(counter, end=", ")
        

18, 27, 36, 45, 54, 63 

**Hints for 1(d):** 

print(5 * (-1)) gives output -5\
print("-" + str(5)) gives output -5

In [4]:
# d)18,-27, 36,-45,54,-63

for counter in range(18,63+1,9):
    if (counter % 2 == 0):
        print(counter, end=", ")
    else:
        if counter == 63:
            print(-1 * (counter), end=" ")
        else:
            print(-1 * (counter), end=", ")


18, -27, 36, -45, 54, -63 

====================================================================\
<font color='blue'>
All the tasks can be solved using both the "while loop" and the "for loop". But you need to choose the best looping construct which is the most suitable for solving a particular task.
</font>

====================================================================

### Task 2
Write a Python code for the following:

1)	Ask the user to enter the name of his favorite car.\
2)	Ask the user to enter a Number

Display the name of the user’s favorite car, the number of times specified in the second step.

**Example:** If the user enters “Toyota” and 20, your program should print the name “Toyota” twenty times.

**Example:** If the user enters “Veyron” and 5, your program should print the name “Veyron” five times.

**Input:**\
Veyron, 5

**Output:**\
Veyron\
Veyron\
Veyron\
Veyron\
Veyron


In [5]:
#Todo

car_name = input("Enter the name of your favourite car: ")
number = int(input("Enter a number: "))

for n in range(number):
    print(car_name)
    

Enter the name of your favourite car: Veyron
Enter a number: 5
Veyron
Veyron
Veyron
Veyron
Veyron


### Task 3
Write a Python code of a program that adds all numbers that are multiples of **both 7 and 9** up to 600 (including 600).

Output: 2835


In [6]:
#Todo

sum = 0

for numbers in range(600+1):
    if numbers % 7 == 0 and numbers % 9 == 0:
        sum += numbers
print(sum)


2835


### Task 4
Write a Python code of a program that adds all numbers that are multiples of **either 7 or 9** up to 600(including 600). \
Ensure that numbers like 63 are added only once in the sum.

Output: 42649


In [7]:
#Todo

sum = 0

for numbers in range(7,600+1):
    if numbers % 7 == 0 or numbers % 9 == 0:
        print(numbers,end =" ")
        sum += numbers
        
print(f"\n\n[Summation of numbers: {sum}]")


7 9 14 18 21 27 28 35 36 42 45 49 54 56 63 70 72 77 81 84 90 91 98 99 105 108 112 117 119 126 133 135 140 144 147 153 154 161 162 168 171 175 180 182 189 196 198 203 207 210 216 217 224 225 231 234 238 243 245 252 259 261 266 270 273 279 280 287 288 294 297 301 306 308 315 322 324 329 333 336 342 343 350 351 357 360 364 369 371 378 385 387 392 396 399 405 406 413 414 420 423 427 432 434 441 448 450 455 459 462 468 469 476 477 483 486 490 495 497 504 511 513 518 522 525 531 532 539 540 546 549 553 558 560 567 574 576 581 585 588 594 595 

[Summation of numbers: 42649]


### Task 5
Write a Python code of a program that adds all numbers that are multiples of **either 7 or 9 but not both**, up to 600(including 600).

Output: 39814


In [8]:
#Todo

sum = 0

for numbers in range(600+1):
    if (numbers % 7 == 0) != (numbers % 9 == 0):
        sum += numbers
print(sum)


39814


### Task 6
Write a Python code to displays all the **odd numbers** between 10 and 50.

Output: 11 13 15 17 19 21 23 25 27 29 31 33 35 37 39 41 43 45 47 49


In [9]:
#Todo

for odd_numbers in range(10,50+1):
    if odd_numbers % 2 != 0:
        print(odd_numbers, end=" ")
        

11 13 15 17 19 21 23 25 27 29 31 33 35 37 39 41 43 45 47 49 

### Task 7
Write a Python code that will calculate the **value of y if the expression** of y is as follows (n is the input):  

$y = 1^2 - 2^2 +3^2 -4^2 +5^2 .........+ n^2 $

**Example:** 

If the user enters 10, y= -55 \
If the user enters 20, y= -210


In [10]:
#Todo

import math

sum = 0
n = int(input("Enter value of n: "))
       
for n in range(1,n+1):
    if (n % 2 == 0):
        sum -= math.pow(n,2)
    else:
        sum += math.pow(n,2)
print("y=",sum)


Enter value of n: 20
y= -210.0


### Task 8
Write a Python code of a program that asks the user to enter ten numbers then display the total and the average of **ONLY** the **odd numbers** among those ten numbers.

**Example(1):** 

Input: 1 2 3 4 5 6 7 8 9 10 \
Output: Total is 25 and Average is 5.0\
(i.e., Total is 25 = (1+3+5+7+9) and Average is 25/5 = 5.0)]


**Example(2):** 

Input: -20  3  -5  40  -17  9  20  -8   99  -200 \
Output: Total is 89 and Average is 17.8 \
(i.e., Total is 89 =(3+(-5)+(-17)+9+99) and Average is 89/5 = 17.8)]




In [11]:
#Todo

sum = 0
count = 0

for counter in range(10):
    number = int(input("Enter numbers:"))
    if number % 2 != 0:
        count += 1
        sum += number
        average = sum / count
        
print(f"Total is {sum} and Average is {average}")


Enter numbers:-20
Enter numbers:3
Enter numbers:-5
Enter numbers:40
Enter numbers:-17
Enter numbers:9
Enter numbers:20
Enter numbers:-8
Enter numbers:99
Enter numbers:-200
Total is 89 and Average is 17.8


## Task 9
Write a Python code for the following:
* Ask the user to enter a Number, N
* Display the summation of multiples of 7 up to that number (**from 1 to N inclusive**)

**Example(1):** If the user **enters 50**, your program should print 196. \
(Because: 7 + 14 + 21 + 28 + 35 + 42 + 49 = 196).

**Example(2):** If the user **enters 75**, your program should print 385. \
(Because: 7 + 14 + 21 + 28 + 35 + 42 + 49 + 56 + 63 + 70 = 385).



In [12]:
#Todo

n = int(input("Enter a Number: "))
sum = 0

for n in range(7,n+1):
    if n % 7 == 0:
        sum += n

print(f"The summation of multiples of 7 is {sum}.")


Enter a Number: 75
The summation of multiples of 7 is 385.


### Task 10
Write a Python code that will read 10 numbers from the user, and then print the first number, the sum of the first 2 numbers, the sum of the first 3 numbers, and so on up to the sum of 10 numbers.

For example,

The user enters 10, output 10, then \
The user enters 2, (10+2) = 12, output 12, then \
The user enters 4, (10+2+4) = 16, output 16, then \
The user enters 20, (10+2+4+20) = 36, output 36, and continuous till the 10$^{th}$ input


In [13]:
#Todo

sum = 0

for i in range(10):
    sum += int(input("Enter number: "))
    print(sum)
    

Enter number: 10
10
Enter number: 2
12
Enter number: 4
16
Enter number: 20
36
Enter number: 23
59
Enter number: 32
91
Enter number: 9
100
Enter number: -50
50
Enter number: 49
99
Enter number: 1
100


### Task 11
Write a Python program which takes a number and prints the digits from the unit place, then the tenth, then hundredth, etc. <font color='red'>(Right to Left)</font>

**Example:** If the user gives 32768, then print 8, 6, 7, 2, 3 

**Hint:** First to get the digit from the right side, take the remainder of the number using modulus (%) operator and print the digit. For dropping the last digit, perform floor division by 10 on the number. Then start over.

32,768 % 10 = 8 

32,768 // 10 = 3, 276 

3, 276 % 10 = 6

3, 276 // 10 = 327 

327 % 10 = 7 

327 // 10 = 32 

32 % 10 = 2 

32 //10 = 3 

3 % 10 = 3 

3 // 10 = 0 

Done! When the number becomes 0 your loop should end.


In [16]:
#Todo

number = int(input("Give a number: "))

while number > 0: 
    number_unit = number % 10
    number //= 10
    
    print(number_unit, end=" ")
    

Give a number: 32768
8 6 7 2 3 

### Task 12

Write a Python program that takes a number and tells how many digits are in that number. 

**Example:** If the user gives 9876, you should print 4. 



**Hint:** Keep floor dividing by ten and count how many times this could be divided. 

9876 floor divide by 10, is 987, count that got 1 digit (total 1) \
987 floor divide by 10, is 98, count that got 1 digit (total 2) \
98 floor divide by 10, is 9, count that got 1 digit (total 3) \
9 floor divide by 10, is 0, count that got 1 digit (total 4) 

Done! (When the number becomes 0 your loop should end.)



In [15]:
#Todo

number = int(input("Give a number: "))
count = 0    
while number > 0: 
    count += 1
    number //= 10
    
print(f"Total digits are in that number is {count}.")


Give a number: 9876
Total digits are in that number is 4.


### Task 13
Write a Python code of a program that takes a number and prints the value of 10 to the power that number. <font color='red'>You need to use a loop because math function power is not allowed in this task.</font> 

For example: if the user gives 3, print 1000. 

**Hint:** Keep multiplying 1 by 10, again and again. \
You need to write, \
$product = product x 10$

$1 x 10 = 10 $ \
$10 x 10 = 100$ \
$100 x 10 = 1000$ 


In [16]:
#Todo

power = int(input("Enter a number: "))
value = 1

for i in range(1,power+1):
    value *= 10
print(f"The value of 10 to the power that number is {value}.")


Enter a number: 3
The value of 10 to the power that number is 1000.


### Task 14

Write a Python program that takes a number from the user and prints the divisors of that number and then print how many divisors were there.

**Example1:** 

Input: 6 \
Output: \
1, 2, 3, 6 \
“Total 4 divisors”.


**Example2:** 

Input: 121 \
Output: \
1, 11, 121

“Total 3 divisors”.



In [17]:
#Todo

number = int(input("Enter a number: "))
i = 1
count = 0

for i in range(i,number+1):
    if number % i == 0 :
        count += 1
        print(i,end=" ") 
         

print(f"\nTotal {count} divisors.")


Enter a number: 121
1 11 121 
Total 3 divisors.


### Task 15
Write a Python program that takes a number and tells if it is a perfect number or not.

**Perfect Number:** An integer number is said to be a perfect number if its factors, including 1 but not the number itself, sum to the number.

**Example1:**

Input: 6 \
Output: “6 is a perfect number” \
Explanation: 6 have 4 divisors: 1, 2, 3, and 6. \
If we add all factors except itself, 6 = 1 + 2 + 3.

**Example2:**

Input: 28 \
Output: “28 is a perfect number” \
Explanation: 28 have 6 divisors: 1, 2, 4, 7, 14, and 28. \
If we add all factors except itself, 28 = 1 + 2 + 4 + 7 + 14.

**Example3:**

Input: 33 \
Output: “33 is not a perfect number” \
Explanation: 33 have 3 divisors: 1, 3, 11, and 33. \
If we add all factors except itself, 15 = 1 + 3 + 11.

In [18]:
#Todo

number = int(input("Enter a number: "))
sum_divisors = 0

for i in range(1,number):
    if number % i == 0:
        sum_divisors = sum_divisors + i
if (sum_divisors == number):
    print(f"{number} is a perfect number.")
else:
    print(f"{number} is not a perfect number.")
    

Enter a number: 33
33 is not a perfect number.


### Task 16

Write a Python program that asks the user for one number and tells if it is a prime number or not. 

**Prime Number:** If a number has only two divisors, (1 and itself), then it is prime. If it is divisible by more numbers, then it is not a prime.

**Hint:** use the divisor count from task 16. 

**Example1:** 

Input: 11 \
Output: “11 is a prime number” \
Explanation: 11 has only 2 divisors: 1, and 11.

**Example2:** 

Input: 6 \
Output: “6 is not a prime number” \
Explanation: 6 have 4 divisors: 1, 2, 3 and 6.




In [5]:
#Todo

number = int(input("Enter a number: "))

if number > 1:
    for i in range(2,number):
        if (number % i) == 0:
            print(f"{number} is not a prime number.")
            break
    else:
        print(f"{number} is a prime number.")
else:
    print(f"{number} is not a prime number.")
    

Enter a number: 5
5 is a prime number.
5 is a prime number.
5 is a prime number.


### Task 17

Write a Python program that asks the user for a quantity then takes that many numbers and prints the maximum, minimum and average of those numbers. 

**Example:**
If the user enters 5 as an input for quantity. Then enters 10, 4, -1, -100, and 1. \
Your program output should be: “Maximum 10”, “Minimum -100”, “Average is -17.2” 

[Average calculation: (10+4 + (-1) + (-100) + 1)/5 = -86/5=-17.2]


In [20]:
#Todo

quantity = int(input("Enter the quantity of numbers: "))

minimum = None
maximum = None
sum = 0
count=0

for i in range(quantity):
    numbers = int(input("Enter number: "))
    count += 1
    sum += numbers
    average = sum / count
    
    if minimum == None or numbers < minimum:
        minimum = numbers
    if maximum == None or numbers > maximum:
        maximum = numbers

print(f"Maximum {maximum}, Minimum {minimum} & Average is {average} of those numbers.")


Enter the quantity of numbers: 5
Enter number: 10
Enter number: 4
Enter number: -1
Enter number: -100
Enter number: 1
Maximum 10, Minimum -100 & Average is -17.2 of those numbers.


### Optional Tasks (18 -21)

### Task 18

Take five numbers from the user and find the minimum and the average of **only the even** numbers entered by the user. [If the user enters odd numbers ignore them]

Assume, the first input is always an even number.

**Example1:** 

If the user enters 10, 4, -1, -100, and 1. \
Output: “Minimum -100”, “Average is 28.66667” \
[Average calculation: (10+4 + (-100))/3 = -86/3=-28.66667]

**Example2:**

If the user enters 2, 10, 1, 21, and 3. \
Output: Minimum 2”, “Average is 6.0” \
Average calculation: [(2+10)/2 = 12/2= 6]

In [26]:
#Todo

minimum = None
sum = 0
count = 0
input_serialno = 1


numbers = int(input(f"Enter the 1st number:"))
    
    
if numbers % 2 == 0:
    
    for i in range(4):
    
        input_serialno += 1
        if input_serialno == 2:
            serialno = "2nd"
        elif input_serialno == 3:
            serialno = "3rd"
        elif input_serialno > 3:
            serialno = str(input_serialno) + "th"  
            
        if numbers % 2 == 0:
            count += 1
            sum += numbers
            average = sum / count
            
            if minimum == None or numbers < minimum:
                minimum = numbers
        
        numbers = int(input(f"Enter the {serialno} number:"))
        
    print(f"Minimum {minimum} & Average is {average} of those numbers.")
    
else:
    print("The first input is always an even number.")


Enter the 1st number:2
Enter the 2nd number:10
Enter the 3rd number:1
Enter the 4th number:21
Enter the 5th number:3
Minimum 2 & Average is 6.0 of those numbers.


### Task 19

Write a Python code for the following:
* Ask the user to enter a number, N (Total number of inputs to be taken). Assume that the user will never enter the first number as zero.
* Take N number of inputs
* Prints out the product of all the numbers read

For example, if the first input is 4, then the program has to read in four numbers from the user and print the product of these four numbers. 

**Example:**\
Input: 5 10 6 4 2 1 \
Output: 480

Explanation:
5 is the total number of inputs taken. Then, the calculation should be 10 X 6 X 4 X 2 X 1 = 480 




In [23]:
#Todo

quantity = int(input("Enter the quantity of numbers: "))
i = 1
input_serialno = 0

if quantity > 0:
    for count in range(quantity):
        
        input_serialno += 1
    
        if input_serialno == 1:
            serialno = "1st"
        elif input_serialno == 2:
            serialno = "2nd"
        elif input_serialno == 3:
            serialno = "3rd"
        elif input_serialno > 3:
            serialno = str(input_serialno)+"th"
        
        numbers = int(input(f"Enter the {serialno} number:"))
    
        i = i * numbers
        
    print(f"\nThe product of these {quantity} numbers is {i}.")
    
else:
    print("You can not input as 0.")
    

Enter the quantity of numbers: 5
Enter the 1st number:10
Enter the 2nd number:6
Enter the 3rd number:4
Enter the 4th number:2
Enter the 5th number:1

The product of these 5 numbers is 480.


#### Task 20
Ask the user for a range (a starting number and an ending number). 
**Count** how many numbers are prime numbers and how many numbers are perfect numbers between that range and **print** those numbers. 

For example, between 2 and 6, there are 3 prime numbers (2, 3, 5) and 1 perfect number (6). 


**Example:**\
Sample Input: 2 6 

Sample Output:

Between 2 and 6,\
Found 3 prime numbers \
Found 1 perfect number.\
Prime numbers: 2, 3, 5.\
Perfect numbers: 6.



In [24]:
#Todo

start = int(input("Enter the starting number:"))
end = int(input("Enter the ending number:"))

for number in range(start,end+1):
    sum = 0
    if number > 1:
        for i in range(2, number):
            if (number % i == 0):
                break
        else:
            print(f"{number} is a prime number.")
        
            
    for i in range(1, number):
        if number % i == 0:   
            sum = sum + i
    if (sum == number):
        print(f"{number} is a perfect number.")

Enter the starting number:2
Enter the ending number:6
2 is a prime number.
3 is a prime number.
5 is a prime number.
6 is a perfect number.


### Task 21
Write a Python program that takes a number from the user and prints its digits from left to right. <font color='red'> (left to right) </font>

**Example:** if the user gives 32768, then print 3, 2, 7, 6, 8 

**Hint:**
* You need to combine Task12 and Task13 for solving this problem.
* Count how many digits are in the input number. <font color='red'>(While counting digits, do not modify the original number)</font> **[Task12]**
* Calculate 10 to the power that number of digits minus 1. **[Task13]**\
(You can reuse the task11 or directly call math.pow(). While using math functions, you need to keep in mind that it returns everything in floating-point numbers.)

Say, 32768 have 5 digits, so you calculate 10 to power 4 which is 10,000. 
Take the remainder of 32,768 by 10,000 and thus you get 2,768 
Then  floor divide 32,768 by 10,000 and thus you get 3. 
Then floor divide 10,000 by 10 to get 1,000 

Take the remainder of 2,768 by 1,000 and thus you get 768.\
Then floor divide 2,768 by 1,000 and thus you get 2.\
Then floor divide 10,00 by 10 to get 1,00 

Keep going on until there are no more digits left (zero!). 

In short:

Loop 1: First count digits; say 5 in this case for 32,768  \
Loop 2: Then calculate 10 to power 4 (5-1), which is 10,000. \
Loop 3: Then repeat following three steps 

32,768 % 10,000 = 2,768
<br>
32,768 // 10,000 = 3 
<br>
10,000//10 = 1,000 
<br><br>

2,768 % 1,000 = 768
<br>
2,768 //1,000 = 2 
<br>
1,000//10 = 100 
<br><br>

768 % 100 = 68 
<br>
768 // 100 = 7 
<br>
100//10 = 10 
<br><br>

68 % 10 = 8
<br>
68 //10 = 6 
<br>
10//10 = 1
<br><br>

8 % 1 = 0
<br>
8 //1 = 8 
<br>
1//10 = 0`


Done! (When the digit count becomes 0, your loop should end.)



In [25]:
#Todo

number = int(input("Enter a number: "))

lenght = (len(str(abs(number)))-1)

for i in range(lenght, -1, -1):
    counter_1 = number // (10 ** i)
    counter_2 = counter_1 % 10
    print(counter_2, end=" ")   

Enter a number: 32768
3 2 7 6 8 